libpromises/evalfunction.c

```diff
@@ -629,34 +692,34 @@ static bool CallHostsSeenCallback(const char *hostkey, const char *address,
     char buf[PRINTSIZE(uintmax_t)];
     xsnprintf(buf, sizeof(buf), "%ju", (uintmax_t) quality->lastseen);
 
-    PrependItem(addresses, address, buf);
-
+    SeqAppend(host_data, HostDataNew(address, hostkey, buf));
     return true;
 }
 
 /*******************************************************************/
 
 static FnCallResult FnCallHostsSeen(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const Policy *policy, ARG_UNUSED const FnCall *fp, const Rlist *finalargs)
 {
-    Item *addresses = NULL;
+    Seq *host_data = SeqNew(1, HostDataFree);
 
     int horizon = IntFromString(RlistScalarValue(finalargs)) * 3600;
     char *hostseen_policy = RlistScalarValue(finalargs->next);
-    char *format = RlistScalarValue(finalargs->next->next);
+    char *format_str = RlistScalarValue(finalargs->next->next);
+    HostsSeenFormatOption format = ValidateHostsSeenFormatOption(format_str);
```

Comment:
```suggestion
    HostsSeenFieldOption format = ValidateHostsSeenFormatOption(field_str);
```

Reply:
i should rename `format` -> `field` too?

Reply:
yep, edited.